### ISSUE TITLE
Lack of accessibility testing

### ISSUE THREAD

**Issue by ghost** [HackFSU/HackFSU-Preview#25]
# Issue

I ran a [simple test](http://wave.webaim.org/report#/hackfsu.com) for accessibility and we have some issues, mostly down to a lack of labels for buttons and other content that would make our site screen reader friendly.  

[![accessibility- issues](https://user-images.githubusercontent.com/14112725/28395150-084b7076-6cc0-11e7-8eb4-277f612aac38.png)](http://wave.webaim.org/report#/hackfsu.com)

# Replication

[Click me!](http://wave.webaim.org/report#/hackfsu.com) The site has an interactive demonstration of the issues (the site doesn't quite render properly so don't worry that it doesn't look exactly as expected).

# Background

The web is built from the ground up with [accessibility standards](https://www.w3.org/standards/webdesign/accessibility) in mind.  Lots of elements have attributes which are totally focused on accessibility: images have alternative text specifically so screen readers can offer a description to visually impaired web surfers.  The official [w3c spec](https://www.w3.org/standards/webdesign/accessibility) is really dense, but luckily there are plenty of people that do this for a living that have lots of useful advice!

- [survival guide](https://www.youtube.com/watch?v=Iu_cUWrWOGM)
- [w3c validator](http://validator.w3.org/check?uri=hackfsu.com&charset=%28detect+automatically%29&doctype=Inline&group=0)
- [code like you give a damn](https://www.youtube.com/watch?v=qdB8SRhqvFc)
- [mdn aria roles introduction](https://developer.mozilla.org/en-US/docs/Web/Accessibility/An_overview_of_accessible_web_applications_and_widgets)

# Steps to Completion

Luckily, many of these problems are already solved (or easily solved), with tools available on npm.  Since we already use [gulp](http://gulpjs.com) we can add something like an accessibility lint.

## Explore using `gulp-accessibility`

[`gulp-accessibility`](https://www.npmjs.com/package/gulp-accessibility) is a lint which catches accessibility errors in our html output, and lets us know where the issues are so we can fix them.  It's relatively easy to install by just following the steps and then adding the snippet suggested in their documentation to start with:

```javascript
gulp.task('test', function() {
  return gulp.src('./example/**/*.html')
    .pipe(access({
      force: true
    }))
    .on('error', console.log)
    .pipe(access.report({reportType: 'txt'}))
    .pipe(rename({
      extname: '.txt'
    }))
    .pipe(gulp.dest('reports/txt'));
});
```

The only problem is that by default it seems to generate a report in a text file, and we would rather have errors appear in the console (like a regular lint).

## AccessLint

AccessLint is an accessibility service that integrates with github, but it's currently under development so it's a little shadowy.  Explore the [site](https://accesslint.com), sign in, and see what's up!

## checklist

- [ ] solve issues marked by [web aim](http://wave.webaim.org/report#/hackfsu.com)
- [ ] use [aria roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/An_overview_of_accessible_web_applications_and_widgets) where appropriate
- [ ] try gulp accessibility
- [ ] explore AccessLint
- [ ] look for other solutions!

# Other thoughts

Should we add accessibility linting to travis-ci to reject pull requests which fail to meet accessibility standards for the front end?

**Comment by ghost** [comment]
@HackFSU/web, thoughts?
